stream_alert/alert_processor/output_base.py

```diff
@@ -0,0 +1,382 @@
+'''
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+import json
+import logging
+import os
+import ssl
+import tempfile
+import urllib2
+
+from collections import namedtuple
+
+import boto3
+from botocore.exceptions import ClientError
+
+from stream_alert.alert_processor.config import load_outputs_config, write_outputs_config
+
+logging.basicConfig()
+LOGGER = logging.getLogger('StreamOutput')
+
+OutputProperty = namedtuple('OutputProperty',
+                            'description, value, is_secret, cred_requirement')
+OutputProperty.__new__.__defaults__ = ('', '', False, False)
+
+
+class OutputRequestFailure(Exception):
+    """OutputRequestFailure handles any HTTP failures"""
+
+
+class OutputBase(object):
+    """StreamOutputBase is the base class to handle routing alters to outputs
```

Comment:
typo: routing alerts*

I would add some more background on how to use this base class, along with going into the class variables just below this comment.  Also describe the public methods and attributes available